spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/StringBasedJdbcQuery.java

```diff
@@ -74,12 +81,14 @@ public class StringBasedJdbcQuery extends AbstractJdbcQuery {
 	private static final String PARAMETER_NEEDS_TO_BE_NAMED = "For queries with named parameters you need to provide names for method parameters; Use @Param for query method parameters, or use the javac flag -parameters";
 	private final JdbcConverter converter;
 	private final RowMapperFactory rowMapperFactory;
-	private final SpelEvaluator spelEvaluator;
+	private final ValueExpressionQueryRewriter.ParsedQuery parsedQuery;
 	private final boolean containsSpelExpressions;
 	private final String query;
 
 	private final CachedRowMapperFactory cachedRowMapperFactory;
 	private final CachedResultSetExtractorFactory cachedResultSetExtractorFactory;
+	private final ValueExpressionDelegate delegate;
+	private final List<Map.Entry<String, String>> parameterBindings;
```

Comment:
`ParsedQuery.getParameterMap()` is now public, I'm going to adopt to this during the merge.